nixos/modules/services/networking/fedimintd.nix

```
@@ -0,0 +1,357 @@
+                {
+                  FM_BIND_P2P = "${cfg.p2p.bind}:${builtins.toString cfg.p2p.port}";
+                  FM_BIND_API = "${cfg.api.bind}:${builtins.toString cfg.api.port}";
+                  FM_P2P_URL = cfg.p2p.address;
+                  FM_API_URL = cfg.api.address;
+                  FM_DATA_DIR = cfg.dataDir;
+                  FM_BITCOIN_NETWORK = cfg.bitcoin.network;
+                  FM_BITCOIN_RPC_URL = cfg.bitcoin.rpc.address;
+                  FM_BITCOIN_RPC_KIND = cfg.bitcoin.rpc.kind;
+                }
+                cfg.extraEnvironment
+              ]);
+              serviceConfig = {
+                User = cfg.user;
+                Group = cfg.group;
+
+                Restart = "always";
+                RestartSec = 10;
+                StartLimitBurst = 5;
+                UMask = "077";
+                LimitNOFILE = "100000";
+
+                LockPersonality = true;
+                ProtectClock = true;
+                ProtectControlGroups = true;
+                ProtectHostname = true;
+                ProtectKernelLogs = true;
+                ProtectKernelModules = true;
+                ProtectKernelTunables = true;
+                PrivateMounts = true;
+                RestrictAddressFamilies = [ "AF_INET" "AF_INET6" ];
+                RestrictNamespaces = true;
+                RestrictRealtime = true;
+                SystemCallArchitectures = "native";
+                SystemCallFilter = [
+                  "@system-service"
+                  "~@privileged"
+                ];
+                StateDirectory = "fedimintd-${fedimintdName}";
+                StateDirectoryMode = "0700";
+                ExecStart = startScript;
+
+                # Hardening measures
+                PrivateTmp = "true";
+                ProtectSystem = "full";
+                NoNewPrivileges = "true";
+                PrivateDevices = "true";
+                MemoryDenyWriteExecute = "true";
+              };
+            }
+          )
+        ))
+        eachFedimintd;
+
+    users.users = mapAttrs'
+      (fedimintdName: cfg: (
+        nameValuePair "fedimintd-${fedimintdName}" {
+          name = cfg.user;
+          group = cfg.group;
+          description = "Fedimint daemon user";
+          home = cfg.dataDir;
+          isSystemUser = true;
+        }
+      ))
+      eachFedimintd;
+
+    users.groups = mapAttrs'
+      (fedimintdName: cfg: (
+        nameValuePair "${cfg.group}" { }
+      ))
+      eachFedimintd;
+
+    services.nginx.virtualHosts = mapAttrs'
+      (fedimintdName: cfg: (
+        nameValuePair cfg.api.fqdn (lib.mkMerge [
+
+          cfg.nginx.config
+
+          {
+            enableACME = mkDefault true;
+            forceSSL = mkDefault true;
+            locations."/ws/" = {
+              proxyPass = "http://127.0.0.1:${builtins.toString cfg.api.port}/";
+              proxyWebsockets = true;
+              extraConfig = "proxy_pass_header Authorization;";
```

```suggestion
              extraConfig = ''
                proxy_pass_header Authorization;
              '';
```
otherwise it might get concatinated wrongly when being merged